integrations/cohere/src/haystack_integrations/components/generators/cohere/chat/chat_generator.py

```diff
@@ -1,37 +1,271 @@
+    tool_plan = ""
+    tool_calls = []
+    current_tool_call = None
+    current_tool_arguments = ""
+    captured_meta = {}
+
+    for chunk in response:
+        if chunk and chunk.type == "content-delta":
+            stream_chunk = StreamingChunk(content=chunk.delta.message.content.text)
+            streaming_callback(stream_chunk)
+            response_text += chunk.delta.message.content.text
+        elif chunk and chunk.type == "tool-plan-delta":
+            tool_plan += chunk.delta.message.tool_plan
+            stream_chunk = StreamingChunk(content=chunk.delta.message.tool_plan)
+            streaming_callback(stream_chunk)
+        elif chunk and chunk.type == "tool-call-start":
+            tool_call = chunk.delta.message.tool_calls
+            current_tool_call = ToolCall(id=tool_call.id, tool_name=tool_call.function.name, arguments="")
+        elif chunk and chunk.type == "tool-call-delta":
+            current_tool_arguments += chunk.delta.message.tool_calls.function.arguments
+        elif chunk and chunk.type == "tool-call-end":
+            if current_tool_call:
+                current_tool_call.arguments = json.loads(current_tool_arguments)
+                tool_calls.append(current_tool_call)
+                current_tool_call = None
+                current_tool_arguments = ""
+        elif chunk and chunk.type == "message-end":
+            captured_meta.update(
+                {
+                    "model": model,
+                    "index": 0,
+                    "finish_reason": chunk.delta.finish_reason,
+                    "usage": {
+                        "prompt_tokens": chunk.delta.usage.billed_units.input_tokens,
+                        "completion_tokens": chunk.delta.usage.billed_units.output_tokens,
+                    },
+                }
+            )
+
+    # Create the appropriate ChatMessage based on what we received
+    if tool_calls:
+        chat_message = ChatMessage.from_assistant(text=tool_plan, tool_calls=tool_calls)
+    else:
+        chat_message = ChatMessage.from_assistant(text=response_text)
+
+    # Add metadata
+    chat_message.meta.update(captured_meta)
+
+    return chat_message
+
+
 @component
 class CohereChatGenerator:
     """
-    Completes chats using Cohere's models through Cohere `chat` endpoint.
+    Completes chats using Cohere's models using Cohere cohere.ClientV2 `chat` endpoint.
 
-    You can customize how the text is generated by passing parameters to the
+    You can customize how the chat response is generated by passing parameters to the
     Cohere API through the `**generation_kwargs` parameter. You can do this when
     initializing or running the component. Any parameter that works with
-     `cohere.Client.chat` will work here too.
+     `cohere.ClientV2.chat` will work here too.
     For details, see [Cohere API](https://docs.cohere.com/reference/chat).
 
     ### Usage example
 
     ```python
+    from haystack import Pipeline
```

Comment:
I suggest we should keep one very short usage example of the CohereChatGenerator without any tools. In addition, we could add a usage example with tools as you suggested. If we do that, we should make sure that all ChatGenerators have consistent usage examples like this. What do you think?